### PR TITLE
Add load balancers to CF

### PR DESCRIPTION
**Pull request by gregturn**
* Model CF routes as load balancers, given routes can be assigned to multiple CF apps (server groups) and used to effectively load balancer between different versions of an app.
* Create op to create a CF load balancer
* Also create op to delete CF load balancer
* Add build info to CF server groups

### CODE REVIEW

**Comment by cfieber** [comment]
seems like this should rebase on top of https://github.com/spinnaker/clouddriver/pull/209 once that goes in (I see some deploy handler changes in this PR that might not be relevant any more?)

**Comment by gregturn** [comment]
Yep. This one will probably require some fixes.

**Comment by gregturn** [comment]
@cfieber @duftler I've rebased this PR against latest master and also overhauled the entire thing so that it really now supports load balancing. There is also https://github.com/spinnaker/deck/pull/1764 that corresponds with setting up load balancers in CF (which is nothing more than a route). Then when deploying using Spinnaker's generic deploy task, this will allow mapping a route to every server group, along with enabling/disabling server groups as desired.

File: kato/kato-cf/src/main/groovy/com/netflix/spinnaker/kato/cf/deploy/validators/AbstractEnableDisableCloudFoundryServerGroupDescriptionValidator.groovy [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2015 Google, Inc.
```
Comment (duftler) [inline kato/kato-cf/src/main/groovy/com/netflix/spinnaker/kato/cf/deploy/validators/AbstractEnableDisableCloudFoundryServerGroupDescriptionValidator.groovy]: s/Google/Cloud Foundry
(Also noticed that a lot of the files don't have copyright headers.)

File: kato/kato-cf/src/main/groovy/com/netflix/spinnaker/kato/cf/deploy/validators/DisableCloudFoundryServerGroupDescriptionValidator.groovy [hunk 2]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2015 Google, Inc.
```
Comment (duftler) [inline kato/kato-cf/src/main/groovy/com/netflix/spinnaker/kato/cf/deploy/validators/DisableCloudFoundryServerGroupDescriptionValidator.groovy]: s/Google/Cloud Foundry

File: kato/kato-cf/src/main/groovy/com/netflix/spinnaker/kato/cf/deploy/validators/EnableCloudFoundryServerGroupDescriptionValidator.groovy [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2015 Google, Inc.
```
Comment (duftler) [inline kato/kato-cf/src/main/groovy/com/netflix/spinnaker/kato/cf/deploy/validators/EnableCloudFoundryServerGroupDescriptionValidator.groovy]: s/Google/Cloud Foundry

File: kato/kato-cf/src/main/groovy/com/netflix/spinnaker/kato/cf/deploy/validators/StandardCfAttributeValidator.groovy [hunk 4]
```diff
@@ -17,7 +17,7 @@
 package com.netflix.spinnaker.kato.cf.deploy.validators
 
 import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
-import com.netflix.spinnaker.kato.cf.security.CloudFoundryAccountCredentials
+import com.netflix.spinnaker.clouddriver.cf.security.CloudFoundryAccountCredentials
```
Comment (duftler) [inline kato/kato-cf/src/main/groovy/com/netflix/spinnaker/kato/cf/deploy/validators/StandardCfAttributeValidator.groovy]: Sort-order is off here.

File: kato/kato-cf/src/main/groovy/com/netflix/spinnaker/kato/config/KatoCloudFoundryConfig.groovy [hunk 5]
```diff
@@ -16,7 +16,7 @@
 package com.netflix.spinnaker.kato.config
 import com.netflix.spinnaker.clouddriver.security.AccountCredentialsRepository
 import com.netflix.spinnaker.kato.cf.deploy.handlers.CloudFoundryDeployHandler
-import com.netflix.spinnaker.kato.cf.security.CloudFoundryAccountCredentials
+import com.netflix.spinnaker.clouddriver.cf.security.CloudFoundryAccountCredentials
```
Comment (duftler) [inline kato/kato-cf/src/main/groovy/com/netflix/spinnaker/kato/config/KatoCloudFoundryConfig.groovy]: Sort-order is off here.

File: kato/kato-cf/src/test/groovy/com/netflix/spinnaker/kato/cf/CloudFoundryOperationsSpec.groovy [hunk 6]
```diff
@@ -21,9 +21,8 @@ import com.netflix.spinnaker.clouddriver.security.AccountCredentials
 import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
 import com.netflix.spinnaker.clouddriver.security.AccountCredentialsRepository
 import com.netflix.spinnaker.clouddriver.security.DefaultAccountCredentialsProvider
-import com.netflix.spinnaker.kato.cf.deploy.description.CloudFoundryDeployDescription
 import com.netflix.spinnaker.kato.cf.deploy.handlers.CloudFoundryDeployHandler
```
Comment (duftler) [inline kato/kato-cf/src/test/groovy/com/netflix/spinnaker/kato/cf/CloudFoundryOperationsSpec.groovy]: Sort-order is off here.

File: kato/kato-cf/src/test/groovy/com/netflix/spinnaker/kato/cf/deploy/converters/CloudFoundryDeployAtomicOperationConverterUnitSpec.groovy [hunk 7]
```diff
@@ -18,7 +18,7 @@ package com.netflix.spinnaker.kato.cf.deploy.converters
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
 import com.netflix.spinnaker.kato.cf.deploy.description.CloudFoundryDeployDescription
-import com.netflix.spinnaker.kato.cf.security.CloudFoundryAccountCredentials
+import com.netflix.spinnaker.clouddriver.cf.security.CloudFoundryAccountCredentials
```
Comment (duftler) [inline kato/kato-cf/src/test/groovy/com/netflix/spinnaker/kato/cf/deploy/converters/CloudFoundryDeployAtomicOperationConverterUnitSpec.groovy]: Sort-order is off here.

File: kato/kato-cf/src/test/groovy/com/netflix/spinnaker/kato/cf/deploy/validators/CloudFoundryDeployDescriptionValidatorSpec.groovy [hunk 8]
```diff
@@ -19,7 +19,7 @@ import com.netflix.spinnaker.clouddriver.security.DefaultAccountCredentialsProvi
 import com.netflix.spinnaker.clouddriver.security.MapBackedAccountCredentialsRepository
 import com.netflix.spinnaker.kato.cf.TestCredential
 import com.netflix.spinnaker.kato.cf.deploy.description.CloudFoundryDeployDescription
-import com.netflix.spinnaker.kato.cf.security.CloudFoundryAccountCredentials
+import com.netflix.spinnaker.clouddriver.cf.security.CloudFoundryAccountCredentials
```
Comment (duftler) [inline kato/kato-cf/src/test/groovy/com/netflix/spinnaker/kato/cf/deploy/validators/CloudFoundryDeployDescriptionValidatorSpec.groovy]: Sort-order is off here.

File: oort/oort-cf/src/main/groovy/com/netflix/spinnaker/oort/cf/model/CloudFoundryApplicationInstanceProvider.groovy [hunk 9]
```diff
@@ -19,7 +19,7 @@ package com.netflix.spinnaker.oort.cf.model
 import com.netflix.spectator.api.Registry
 import com.netflix.spectator.api.Timer
 import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
-import com.netflix.spinnaker.kato.cf.security.CloudFoundryAccountCredentials
+import com.netflix.spinnaker.clouddriver.cf.security.CloudFoundryAccountCredentials
```
Comment (duftler) [inline oort/oort-cf/src/main/groovy/com/netflix/spinnaker/oort/cf/model/CloudFoundryApplicationInstanceProvider.groovy]: Sort-order is off here.

**Comment by duftler** [comment]
Other than the nits, anything remaining to be done here? If you want to address those minor comments and rebase, I'll get this merged in.

**Comment by duftler** [comment]
LGTM.

**Pull request merged**